common/models/transducer/rna_pure_tf.py

```diff
@@ -0,0 +1,641 @@
+#!/usr/bin/env python3
+# vim: sw=2
+"""
+Implementation of the RNA loss in pure TF,
+plus comparisons against reference implementations.
+This is very similar to RNN-T loss, but restricts
+the paths to be strictly monotonic.
+references:
+  * recurrent neural aligner:
+      https://pdfs.semanticscholar.org/7703/a2c5468ecbee5b62c048339a03358ed5fe19.pdf
+"""
+import os
+import sys
+from returnn.tf.compat import v1 as tf
+from returnn.tf.util.data import Data
+
+NEG_INF = -float("inf")
+
+
+def py_print_iteration_info(msg, var, n, *vars, debug=True):
+  """adds a tf.print op to the graph while ensuring it will run (when the output is used)."""
+  if not debug:
+    return var
+  var_print = tf.print("n=", n, "\t", msg, tf.shape(var), var, *vars,
+                       summarize=-1, output_stream=sys.stdout)
+  with tf.control_dependencies([var_print]):
+    var = tf.identity(var)
+  return var
+
+
+def compute_alignment_tf(bt_mat, input_lens, label_lens):
```

Comment:
This is used for to calculate both the `rna_loss` and the `rna_alignment`

Reply:
No? This pretty much looks like it calculates the alignment. Also the docstring says this.